cob_twist_controller/include/cob_twist_controller/limiters/limiter.h

```diff
@@ -35,145 +36,203 @@
 
 #define LIMIT_SAFETY_THRESHOLD 0.1/180.0*M_PI
 
+/* BEGIN LimiterJointContainer *******************************************************************************/
+/// Joint Container for limiters, implementing interface methods.
+class LimiterJointContainer : public LimiterJointBase
+{
+    public:
+        /**
+         * Specific implementation of enforceLimits-method.
+         * See base class LimiterJointBase for more details on params and returns.
+         */
+        virtual KDL::JntArray enforceLimits(const KDL::JntArray& q_dot_ik, const KDL::JntArray& q) const;
+
+        /**
+         * Initialization for the container.
+         */
+        void init();
+
+        virtual ~LimiterJointContainer();
+
+        explicit LimiterJointContainer(const LimiterParams& limiter_params)
+            : LimiterJointBase(limiter_params)
+        {}
+
+    protected:
+        std::vector<const LimiterJointBase*> limiters_;
+        typedef std::vector<const LimiterJointBase*>::const_iterator LimIter_t;
+
+        /**
+         * Add method
+         * @param lb An implementation of a limiter.
+         */
+        void add(const LimiterJointBase* lb);
+
+        /**
+         * Erase all
+         */
+        void eraseAll();
+};
+/* END LimiterJointContainer *****************************************************************************************/
+
 /* BEGIN LimiterContainer *******************************************************************************/
 /// Container for limiters, implementing interface methods.
-class LimiterContainer : public LimiterBase
+class LimiterCartesianContainer : public LimiterCartesianBase
 {
     public:
         /**
          * Specific implementation of enforceLimits-method.
-         * See base class LimiterBase for more details on params and returns.
+         * See base class LimiterCartesianBase for more details on params and returns.
          */
-        virtual KDL::JntArray enforceLimits(const KDL::JntArray& q_dot_ik, const KDL::JntArray& q) const;
+
+        virtual KDL::Twist enforceCartesianLimits(const KDL::Twist& v_in) const;
```

Comment:
keep the function name `enforceLimits` - no need for the "Cartesian in the function name